### DISCUSSION TITLE
Is there a way to make the Y axis only display whole numbers instead of decimals?

### DISCUSSION THREAD

**Discussion by ErickSorto** [patrykandpatrick/vico#514] [Questions] [answered]
I want to customize the y-axis to contain only whole numbers. I tried tweaking `axisValuesOverrider` but this only scales the y-axis and it still displays decimal numbers. I want to show whole numbers no matter how big the biggest bar is.

**Comment by ErickSorto** [comment, marked as answer]
I figured it out!

```Kotlin
// Create a column chart
chart = columnChart(
    columns = listColumn,
    axisValuesOverrider = AxisValuesOverrider.fixed(
        minY = 0f, // Set minimum Y-axis value
        maxY = entries.maxOf {
            if (it.y % 2 == 0f) it.y else it.y + 1f // Adjust Y-axis value for even numbers
        }
    ),
    chartModelProducer = chartEntryModelProducer,
    startAxis = rememberStartAxis().apply {
        itemPlacer =
            AxisItemPlacer.Vertical.default(
                5,  //Maximum of 5 Y-axis labels (choose an odd number)
            )
    }
)
```

**Reply by patrickmichalik** [reply]
Hello! I’m glad you’ve found a solution. The approach looks good, but please note that when using a `ChartEntryModelProducer`, you shouldn’t directly use the `ChartEntry` lists passed to `setEntries` and `setEntriesSuspending` for chart configuration. Because `ChartEntryModelProducer`s handle updates asynchronously, doing so may lead to synchronization issues, which in turn can cause numerous glitches. This is particularly important when it comes to `AxisValuesOverrider`s. I’d suggest the following:

```kt
remember {
    object : AxisValuesOverrider<ChartEntryModel> {
        override fun getMaxY(model: ChartEntryModel) = model.maxY + model.maxY % 2
    }
}
```

Also, there’s no need to use `apply` with `rememberStartAxis`, and it would be a good idea to use `remember` for the `AxisItemPlacer.Vertical.default` call:

```kt
rememberStartAxis(itemPlacer = remember { AxisItemPlacer.Vertical.default(maxItemCount = 5) })
```
